### DISCUSSION TITLE
Dependency analysis of deno

### DISCUSSION THREAD

**Discussion by 0x1af2aec8f957** [denoland/deno#12132] [Q&A] [answered]
Is there a command like `npm ls` in `Deno` to view dependencies?
Recently, when I used `Deno` to build a `React` application, I encountered the problem that external dependent packages refer to different versions of react many times, but I didn't find the relevant command to view the dependent package reference analysis in `Deno`.

**Comment by kitsonk** [comment, marked as answer]
`deno info <entrypoint>` is what you are looking for.

**Reply by kitsonk** [reply]
Manual link as well: https://deno.land/manual/tools/dependency_inspector

**Reply by 0x1af2aec8f957** [reply]
Thank you for your reply, which solves my problem.
I would like to know whether to consider transplanting `npm ls` related options later?
For tool developers, many things are necessary, such as output in `JSON` format.

```bash
npm ls

List installed packages

Usage:
npm ls [[<@scope>/]<pkg> ...]

Options:
[-a|--all] [--json] [-l|--long] [-p|--parseable] [-g|--global] [--depth <depth>]
[--omit <dev|optional|peer> [--omit <dev|optional|peer> ...]] [--link]
[--package-lock-only] [--no-unicode]
[-w|--workspace <workspace-name> [-w|--workspace <workspace-name> ...]]
[-ws|--workspaces]

alias: list
```

**Reply by crowlKats** [reply]
>  For tool developers, many things are necessary, such as output in `JSON` format.

that is already possible with the `--json` flag

**Reply by kitsonk** [reply]
`deno info` supports a `--info` flag already. You can see information about commands via `deno help <subcommand>`. The JSON Schema is available here: https://github.com/denoland/deno/blob/main/cli/schemas/module-graph.json.

In addition, there is https://deno.land/x/deno_graph which can provide a runtime API with similar information that is a standalone version of the dependency logic compiled to WASM, exposing a JavaScript/TypeScript interface, that we are refactoring back into the code base. `deno doc` already uses it.
